flang/test/Integration/mixed-lang-stdlib.cpp

```diff
@@ -0,0 +1,48 @@
+// REQUIRES: system-linux
```

Comment:
I'm not sure about this test. Usually we would go about testing something like this by running FileCheck on the output of `flang-new -###` to see what is passed to the linker (and that the linker gets called at all instead of us generating an error for an invalid argument).

Reply:
As I stated in my other comment, the `-###` output is not affected by the `-stdlib=` flag, so a test which looks at its output would not provide a great value. The integration test from the other hand guards that the real-life scenario that I've encountered could be resolved with `flang-new`.